DayZ/handlers/players/server/damage_player.lua

```diff
@@ -241,21 +241,33 @@ function rearmPlayerWeapon (weaponName,slot)
 	takeAllWeapons(source)
 	--Rearm
 	local ammoData,weapID = getWeaponAmmoFromName(weaponName)
-	if getElementData(source,ammoData) <= 0 then 
-		triggerClientEvent (source, "displayClientInfo", source,"Rearm",shownInfos["nomag"],255,22,0) 
-		return 
+	if not (ammoData == "others") then
+		if getElementData(source,ammoData) <= 0 then 
+			triggerClientEvent (source, "displayClientInfo", source,"Rearm",shownInfos["nomag"],255,22,0) 
+			return 
+		end
 	end
 	playerStatusTable[source]["currentweapon_"..slot] = weaponName
 	--Old Weapons
 	local weapon = playerStatusTable[source]["currentweapon_1"]
 	if weapon then
 		local ammoData,weapID = getWeaponAmmoFromName (weapon)
-		giveWeapon(source,weapID,getElementData(source,ammoData), true )
+		if (ammoData == "others") then
+				giveWeapon(source,weapID,1, true )
+		else 
+			local ammoData,weapID = getWeaponAmmoFromName (weapon)
+			giveWeapon(source,weapID,getElementData(source,ammoData), true )
+		end
```

Comment:
I was going to comment about indentation, but you've done a great work here. Is it tested?

Reply:
Yeah, I tested it for 20minutes before I made the commit, then notified people in the discord server that they may want to test it, and it seemed to work for them too. Also did a much longer test run yesterday, played with a friend for like 3 hours. Works like a charm, although equipping the hatchet is still a bit weird. It works, but it's weird. Probably something to do with it being the only melee weapon to use the Carrying/Holding-functionality.

Reply:
I have a server running at the moment, so I can give you the address on discord if you'd like to test it too :)

Reply:
I'm gonna join it soon ;)